pixels-index/pixels-index-rocksdb/src/main/java/io/pixelsdb/pixels/index/rocksdb/RocksDBIndex.java

```diff
@@ -454,4 +516,11 @@ private long extractRowIdFromKey(byte[] keyBytes, int prefixLength)
         // Convert rowId to long
         return ByteBuffer.wrap(rowIdBytes).getLong();
     }
+
+    // Update rowId for single point index
+    private void overwriteRowId(byte[] key, long value)
+    {
+        byte[] minusOneBytes = ByteBuffer.allocate(Long.BYTES).putLong(value).array();
```

Comment:
Directly overwrite bytes in the key instead of allocating a new byte buffer. This is a bit dangerous but worthy. Add a notice in the Java doc saying this method overwrites the content of the input key.